sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayDataTest.java

```diff
@@ -743,21 +824,90 @@ public void populateDisplayData(Builder builder) {
       });
   }
 
+  @Test
   public void testAcceptsNullOptionalValues() {
     DisplayData.from(
       new HasDisplayData() {
         @Override
         public void populateDisplayData(Builder builder) {
           builder.add("key", "value")
               .withLabel(null)
-              .withLinkUrl(null)
-              .withNamespace(null);
+              .withLinkUrl(null);
         }
       });
 
     // Should not throw
   }
 
+  @Test
+  public void testJsonSerialization() throws IOException {
+    HasDisplayData component = new HasDisplayData() {
+      @Override
+      public void populateDisplayData(Builder builder) {
+        builder
+            .add("string", "foobar")
+            .add("long", 1234)
+            .add("double", 123.4)
+            .add("boolean", true)
+            .add("instant", new Instant(0))
+            .add("duration", Duration.millis(1234))
+            .add("class", DisplayDataTest.class)
+              .withLinkUrl("http://abc")
+              .withLabel("baz")
+        ;
+      }
+    };
+    DisplayData data = DisplayData.from(component);
+
+    JsonNode json = MAPPER.readTree(MAPPER.writeValueAsBytes(data));
+    assertThat(json, hasItem(expectedJson(component.getClass(), "STRING", "string", "\"foobar\"")));
```

Comment:
This test may be easier to read (and verify it matches expectations) if it just has the expected value as a single string, or a string for each value.